File: app/src/main/java/net/idik/lib/slimadapter/example/MainActivity.java
package net.idik.lib.slimadapter.example;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.Toast;

import net.idik.lib.slimadapter.SlimInjector;
import net.idik.lib.slimadapter.SlimAdapter;
import net.idik.lib.slimadapter.viewinjector.IViewInjector;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

public class MainActivity extends AppCompatActivity {

    private RecyclerView recyclerView;

    private static final List<Object> data = new ArrayList<>();

    static {
        data.add("hello");
        data.add(",");
        data.add(666666);
        data.add("world");
        data.add("!");
        data.add(666666);
        data.add(666669L);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        recyclerView = (RecyclerView) findViewById(R.id.recyler_view);
        recyclerView.setLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false));
        SlimAdapter.create()
                .withDefault(R.layout.item_string, new SlimInjector() {
                    @Override
                    protected void onInject(Object data, IViewInjector injector) {
                        injector.text(R.id.text, data.toString())
                                .clicked(R.id.text, new View.OnClickListener() {
                                    @Override
                                    public void onClick(View v) {
                                        Toast.makeText(MainActivity.this, "DEFAULT INJECT", Toast.LENGTH_LONG).show();
                                    }
                                });

                    }
                })
                .with(R.layout.item_interger, new SlimInjector<Integer>() {
                    @Override
                    protected void onInject(Integer data, IViewInjector injector) {
                        injector.text(R.id.text, data.toString());

                    }
                })
                .with(R.layout.item_interger, new SlimInjector<String>() {
                    @Override
                    protected void onInject(String data, IViewInjector injector) {
                        injector.text(R.id.text, data);
                    }
                })
                .with(R.layout.item_string, new SlimInjector<CharSequence>() {
                    @Override
                    protected void onInject(CharSequence data, IViewInjector injector) {
                        injector.text(R.id.text, data);
                    }
                })
                .attachTo(recyclerView)
                .setData(data)
                .notifyDataSetChanged();
    }
}

File: exampleforkotlin/src/main/java/net/idik/lib/slimadapter/exampleforkotlin/MainActivity.kt
package net.idik.lib.slimadapter.exampleforkotlin

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.annotation.IntegerRes
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.widget.Toast
import net.idik.lib.slimadapter.SlimAdapter

import java.util.ArrayList

class MainActivity : AppCompatActivity() {
    private val recyclerView: RecyclerView by lazy<RecyclerView> {
        (findViewById(R.id.recyler_view) as RecyclerView).apply {
            layoutManager = LinearLayoutManager(this@MainActivity, LinearLayoutManager.VERTICAL, false)
        }
    }

    private val adapter by lazy {
        SlimAdapter.create()
                .register<String>(R.layout.item_string) { data, injector ->
                    injector.text(R.id.text, data)
                }
                .register<User>(R.layout.item_user) { data, injector ->
                    injector.text(R.id.name, data.name)
                            .text(R.id.age, data.age.toString())
                            .clicked(R.id.name) {
                                Toast.makeText(this@MainActivity, "click user name", Toast.LENGTH_LONG).show()
                            }
                }
                .register<Int>(R.layout.item_interger) { data, injector ->
                    injector.text(R.id.text, data.toString())
                            .longClicked(R.id.text) {
                                Toast.makeText(this@MainActivity, "longclick int", Toast.LENGTH_LONG).show()
                                true
                            }
                }
                .registerDefault(R.layout.item_string) { data, injector ->
                    injector.text(R.id.text, data.toString())
                            .clicked(R.id.text) {
                                Toast.makeText(this@MainActivity, "click default", Toast.LENGTH_LONG).show()
                            }

                }
                .attachTo(recyclerView)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        adapter.setData(data).notifyDataSetChanged()
    }

    companion object {

        private val data = ArrayList<Any>()

        init {
            data.add(666666)
            data.add("hello")
            data.add(",")
            data.add(User("iDIK", 27))
            data.add("world")
            data.add("with")
            data.add("kotlin")
            data.add("!")
            data.add(34234)
            data.add(666669L)
        }
    }
}
